src/compiler/checker.ts

```diff
@@ -4973,6 +4973,16 @@ namespace ts {
             return links.declaredType;
         }
 
+        function isStringConcatExpression(expr: Node): boolean {
+            if (expr.kind === SyntaxKind.StringLiteral) {
+                return true;
+            }
+            else if (expr.kind === SyntaxKind.BinaryExpression) {
+                return isStringConcatExpression((<BinaryExpression>expr).left) && isStringConcatExpression((<BinaryExpression>expr).right);

```

Comment:
I'll be pedantic even though this is only part of what determines whether an enum type is a literal enum type: but also need to check whether the token is a `PlusToken`